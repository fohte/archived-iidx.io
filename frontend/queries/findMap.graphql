query findMap($id: ID!, $playStyle: PlayStyle!, $difficulty: Difficulty!) {
  music(id: $id) {
    id
    title
    subTitle
    genre
    artist
    textageUid
    series
    leggendaria
    map(playStyle: $playStyle, difficulty: $difficulty) {
      id
      numNotes
      level
      playStyle
      difficulty
      minBpm
      maxBpm
    }
  }
}
