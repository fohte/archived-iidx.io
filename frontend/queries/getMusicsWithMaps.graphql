query getMusicsWithMaps {
  musics {
    __typename
    id
    title
    subTitle
    maps {
      __typename
      id
      difficulty
      level
      playStyle
    }
  }
}
