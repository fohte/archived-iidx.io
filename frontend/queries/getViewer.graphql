query getViewer {
  viewer {
    __typename
    id
    uid
  }
}
