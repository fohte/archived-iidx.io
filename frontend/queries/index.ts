/* tslint:disable */

export interface Query {
  musics?: Music[] | null /** Find musics. */
  user?: User | null /** Find a user by ID. */
  viewer?: User | null /** The currently authenticated user. */
}

export interface Music {
  artist: string
  genre: string
  id: string
  leggendaria: boolean
  maps: Map[]
  series: number
  subTitle: string
  textageUid: string
  title: string
}

export interface Map {
  difficulty: Difficulty
  id: string
  level: number
  maxBpm: number
  minBpm: number
  numNotes: number
  playStyle: PlayStyle
}

export interface User {
  id: string
  profile: UserProfile
  uid: string
}

export interface UserProfile {
  name: string
}

export interface Mutation {}
export interface UserQueryArgs {
  id: string
}

export enum Difficulty {
  NORMAL = 'NORMAL',
  HYPER = 'HYPER',
  ANOTHER = 'ANOTHER',
}

export enum PlayStyle {
  SP = 'SP',
  DP = 'DP',
}

export type GetMusicsWithMapsVariables = {}

export type GetMusicsWithMapsQuery = {
  __typename?: 'Query'
  musics?: GetMusicsWithMapsMusics[] | null
}

export type GetMusicsWithMapsMusics = {
  __typename?: 'Music'
  id: string
  title: string
  subTitle: string
  maps: GetMusicsWithMapsMaps[]
}

export type GetMusicsWithMapsMaps = {
  __typename?: 'Map'
  id: string
  difficulty: Difficulty
  level: number
  playStyle: PlayStyle
}

export type GetViewerVariables = {}

export type GetViewerQuery = {
  __typename?: 'Query'
  viewer?: GetViewerViewer | null
}

export type GetViewerViewer = {
  __typename?: 'User'
  id: string
  uid: string
}

import * as ReactApollo from 'react-apollo'

export class GetMusicsWithMapsComponent extends ReactApollo.Query<
  Query,
  GetMusicsWithMapsVariables
> {}
export class GetViewerComponent extends ReactApollo.Query<
  Query,
  GetViewerVariables
> {}
