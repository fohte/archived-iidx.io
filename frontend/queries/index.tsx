export type Maybe<T> = T | null

export enum PlayStyle {
  Sp = 'SP',
  Dp = 'DP',
}

export enum Difficulty {
  Normal = 'NORMAL',
  Hyper = 'HYPER',
  Another = 'ANOTHER',
}

export enum ClearLamp {
  Failed = 'FAILED',
  Assist = 'ASSIST',
  Easy = 'EASY',
  Normal = 'NORMAL',
  Hard = 'HARD',
  ExHard = 'EX_HARD',
  FullCombo = 'FULL_COMBO',
}

export enum Grade {
  A = 'A',
  Aa = 'AA',
  Aaa = 'AAA',
  B = 'B',
  C = 'C',
  D = 'D',
  E = 'E',
  F = 'F',
}

// ====================================================
// Documents
// ====================================================

export type FindMapVariables = {
  id: string
  playStyle: PlayStyle
  difficulty: Difficulty
  username: string
}

export type FindMapQuery = {
  __typename?: 'Query'

  music: Maybe<FindMapMusic>
}

export type FindMapMusic = {
  __typename?: 'Music'

  id: string

  title: string

  subTitle: string

  genre: string

  artist: string

  textageUid: string

  series: number

  leggendaria: boolean

  map: Maybe<FindMapMap>
}

export type FindMapMap = {
  __typename?: 'Map'

  id: string

  numNotes: number

  level: number

  playStyle: PlayStyle

  difficulty: Difficulty

  minBpm: number

  maxBpm: number

  bestResult: Maybe<FindMapBestResult>
}

export type FindMapBestResult = {
  __typename?: 'Result'

  id: string

  score: number

  missCount: number

  clearLamp: ClearLamp

  grade: Grade
}

export type FindUserVariables = {
  screenName: string
}

export type FindUserQuery = {
  __typename?: 'Query'

  user: Maybe<FindUserUser>
}

export type FindUserUser = {
  __typename: 'User'

  name: string
}

export type GetMusicsWithMapsVariables = {}

export type GetMusicsWithMapsQuery = {
  __typename?: 'Query'

  musics: Maybe<GetMusicsWithMapsMusics[]>
}

export type GetMusicsWithMapsMusics = {
  __typename: 'Music'

  id: string

  title: string

  subTitle: string

  maps: GetMusicsWithMapsMaps[]
}

export type GetMusicsWithMapsMaps = {
  __typename: 'Map'

  id: string

  difficulty: Difficulty

  level: number

  playStyle: PlayStyle
}

export type GetViewerVariables = {}

export type GetViewerQuery = {
  __typename?: 'Query'

  viewer: Maybe<GetViewerViewer>
}

export type GetViewerViewer = {
  __typename: 'User'

  id: string

  name: string
}

export type RegisterVariables = {
  username: string
  displayName?: Maybe<string>
}

export type RegisterMutation = {
  __typename?: 'Mutation'

  createUser: Maybe<RegisterCreateUser>
}

export type RegisterCreateUser = {
  __typename?: 'CreateUserPayload'

  user: RegisterUser
}

export type RegisterUser = {
  __typename?: 'User'

  name: string

  profile: RegisterProfile
}

export type RegisterProfile = {
  __typename?: 'UserProfile'

  displayName: string
}

export type RegisterResultsWithCsvVariables = {
  csv: string
  playStyle: PlayStyle
}

export type RegisterResultsWithCsvMutation = {
  __typename?: 'Mutation'

  registerResultsWithCSV: Maybe<RegisterResultsWithCsvRegisterResultsWithCsv>
}

export type RegisterResultsWithCsvRegisterResultsWithCsv = {
  __typename?: 'RegisterResultsWithCSVPayload'

  success: boolean
}

import * as ReactApollo from 'react-apollo'
import * as React from 'react'

import gql from 'graphql-tag'

// ====================================================
// Components
// ====================================================

export const FindMapDocument = gql`
  query findMap(
    $id: ID!
    $playStyle: PlayStyle!
    $difficulty: Difficulty!
    $username: String!
  ) {
    music(id: $id) {
      id
      title
      subTitle
      genre
      artist
      textageUid
      series
      leggendaria
      map(playStyle: $playStyle, difficulty: $difficulty) {
        id
        numNotes
        level
        playStyle
        difficulty
        minBpm
        maxBpm
        bestResult(username: $username) {
          id
          score
          missCount
          clearLamp
          grade
        }
      }
    }
  }
`
export class FindMapComponent extends React.Component<
  Partial<ReactApollo.QueryProps<FindMapQuery, FindMapVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<FindMapQuery, FindMapVariables>
        query={FindMapDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type FindMapProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<FindMapQuery, FindMapVariables>
> &
  TChildProps
export function FindMapHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        FindMapQuery,
        FindMapVariables,
        FindMapProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    FindMapQuery,
    FindMapVariables,
    FindMapProps<TChildProps>
  >(FindMapDocument, operationOptions)
}
export const FindUserDocument = gql`
  query findUser($screenName: String!) {
    user(name: $screenName) {
      __typename
      name
    }
  }
`
export class FindUserComponent extends React.Component<
  Partial<ReactApollo.QueryProps<FindUserQuery, FindUserVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<FindUserQuery, FindUserVariables>
        query={FindUserDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type FindUserProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<FindUserQuery, FindUserVariables>
> &
  TChildProps
export function FindUserHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        FindUserQuery,
        FindUserVariables,
        FindUserProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    FindUserQuery,
    FindUserVariables,
    FindUserProps<TChildProps>
  >(FindUserDocument, operationOptions)
}
export const GetMusicsWithMapsDocument = gql`
  query getMusicsWithMaps {
    musics {
      __typename
      id
      title
      subTitle
      maps {
        __typename
        id
        difficulty
        level
        playStyle
      }
    }
  }
`
export class GetMusicsWithMapsComponent extends React.Component<
  Partial<
    ReactApollo.QueryProps<GetMusicsWithMapsQuery, GetMusicsWithMapsVariables>
  >
> {
  render() {
    return (
      <ReactApollo.Query<GetMusicsWithMapsQuery, GetMusicsWithMapsVariables>
        query={GetMusicsWithMapsDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type GetMusicsWithMapsProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<GetMusicsWithMapsQuery, GetMusicsWithMapsVariables>
> &
  TChildProps
export function GetMusicsWithMapsHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        GetMusicsWithMapsQuery,
        GetMusicsWithMapsVariables,
        GetMusicsWithMapsProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    GetMusicsWithMapsQuery,
    GetMusicsWithMapsVariables,
    GetMusicsWithMapsProps<TChildProps>
  >(GetMusicsWithMapsDocument, operationOptions)
}
export const GetViewerDocument = gql`
  query getViewer {
    viewer {
      __typename
      id
      name
    }
  }
`
export class GetViewerComponent extends React.Component<
  Partial<ReactApollo.QueryProps<GetViewerQuery, GetViewerVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<GetViewerQuery, GetViewerVariables>
        query={GetViewerDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type GetViewerProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<GetViewerQuery, GetViewerVariables>
> &
  TChildProps
export function GetViewerHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        GetViewerQuery,
        GetViewerVariables,
        GetViewerProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    GetViewerQuery,
    GetViewerVariables,
    GetViewerProps<TChildProps>
  >(GetViewerDocument, operationOptions)
}
export const RegisterDocument = gql`
  mutation register($username: String!, $displayName: String) {
    createUser(username: $username, displayName: $displayName) {
      user {
        name
        profile {
          displayName
        }
      }
    }
  }
`
export class RegisterComponent extends React.Component<
  Partial<ReactApollo.MutationProps<RegisterMutation, RegisterVariables>>
> {
  render() {
    return (
      <ReactApollo.Mutation<RegisterMutation, RegisterVariables>
        mutation={RegisterDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type RegisterProps<TChildProps = any> = Partial<
  ReactApollo.MutateProps<RegisterMutation, RegisterVariables>
> &
  TChildProps
export type RegisterMutationFn = ReactApollo.MutationFn<
  RegisterMutation,
  RegisterVariables
>
export function RegisterHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        RegisterMutation,
        RegisterVariables,
        RegisterProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    RegisterMutation,
    RegisterVariables,
    RegisterProps<TChildProps>
  >(RegisterDocument, operationOptions)
}
export const RegisterResultsWithCsvDocument = gql`
  mutation registerResultsWithCSV($csv: String!, $playStyle: PlayStyle!) {
    registerResultsWithCSV(csv: $csv, playStyle: $playStyle) {
      success
    }
  }
`
export class RegisterResultsWithCsvComponent extends React.Component<
  Partial<
    ReactApollo.MutationProps<
      RegisterResultsWithCsvMutation,
      RegisterResultsWithCsvVariables
    >
  >
> {
  render() {
    return (
      <ReactApollo.Mutation<
        RegisterResultsWithCsvMutation,
        RegisterResultsWithCsvVariables
      >
        mutation={RegisterResultsWithCsvDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type RegisterResultsWithCsvProps<TChildProps = any> = Partial<
  ReactApollo.MutateProps<
    RegisterResultsWithCsvMutation,
    RegisterResultsWithCsvVariables
  >
> &
  TChildProps
export type RegisterResultsWithCsvMutationFn = ReactApollo.MutationFn<
  RegisterResultsWithCsvMutation,
  RegisterResultsWithCsvVariables
>
export function RegisterResultsWithCsvHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        RegisterResultsWithCsvMutation,
        RegisterResultsWithCsvVariables,
        RegisterResultsWithCsvProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    RegisterResultsWithCsvMutation,
    RegisterResultsWithCsvVariables,
    RegisterResultsWithCsvProps<TChildProps>
  >(RegisterResultsWithCsvDocument, operationOptions)
}
