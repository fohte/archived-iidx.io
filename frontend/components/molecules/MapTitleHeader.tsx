import * as React from 'react'
import { Header, List } from 'semantic-ui-react'

import { generateTextageURL } from '@app/lib/textage'
import { FindMapMap, FindMapMusic, PlayStyle } from '@app/queries'

export type Props = {
  music: FindMapMusic
  map: FindMapMap
}

const printBPM = ({ minBpm, maxBpm }: FindMapMap) =>
  minBpm === maxBpm ? `${maxBpm}` : `${minBpm}-${maxBpm}`

const MapTitleHeader: React.SFC<Props> = ({ music, map }) => (
  <>
    <div>
      {music.genre}
      <Header as="h2">
        {music.title} {music.subTitle}
      </Header>
      {music.artist}
    </div>

    <div>
      <List divided horizontal>
        <List.Item>
          {map.playStyle} {map.difficulty}
        </List.Item>
        <List.Item>â˜†{map.level}</List.Item>
        <List.Item>BPM {printBPM(map)}</List.Item>
        <List.Item>{map.numNotes} notes</List.Item>
      </List>
    </div>

    <div>
      <span style={{ marginRight: '0.5em' }}>TexTage:</span>
      <List divided horizontal>
        {map.playStyle === PlayStyle.Sp ? (
          [1, 2].map((playSide: 1 | 2) => (
            <List.Item key={playSide}>
              <a
                href={generateTextageURL(music, map, {
                  playStyle: PlayStyle.Sp,
                  playSide,
                })}
              >{`${playSide}P`}</a>
            </List.Item>
          ))
        ) : (
          <List.Item>
            <a
              href={generateTextageURL(music, map, {
                playStyle: PlayStyle.Dp,
              })}
            >
              DP
            </a>
          </List.Item>
        )}
      </List>
    </div>
  </>
)

export default MapTitleHeader
