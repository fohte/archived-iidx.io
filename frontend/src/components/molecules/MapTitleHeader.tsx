import * as React from 'react'
import { Header, List } from 'semantic-ui-react'

import { generateTextageURL } from '@app/lib/textage'
import { Difficulty, PlayStyle } from '@app/queries'

export interface Music {
  title: string
  genre: string
  artist: string
  series: number
  textageUid: string
}

export interface Map {
  level: number
  playStyle: PlayStyle
  difficulty: Difficulty
  minBpm: number
  maxBpm: number
  numNotes: number
}

export type Props = {
  music: Music
  map: Map
}

const printBPM = ({ minBpm, maxBpm }: Map) =>
  minBpm === maxBpm ? `${maxBpm}` : `${minBpm}-${maxBpm}`

const MapTitleHeader: React.SFC<Props> = ({ music, map }) => (
  <>
    <div>
      {music.genre}
      <Header as="h2">{music.title}</Header>
      {music.artist}
    </div>

    <div>
      <List divided horizontal>
        <List.Item>
          {map.playStyle} {map.difficulty}
        </List.Item>
        <List.Item>â˜†{map.level}</List.Item>
        <List.Item>BPM {printBPM(map)}</List.Item>
        <List.Item>{map.numNotes} notes</List.Item>
      </List>
    </div>

    <div>
      <span style={{ marginRight: '0.5em' }}>TexTage:</span>
      <List divided horizontal>
        {map.playStyle === PlayStyle.Sp ? (
          [1, 2].map((playSide: 1 | 2) => (
            <List.Item key={playSide}>
              <a
                href={generateTextageURL(music, map, {
                  playStyle: PlayStyle.Sp,
                  playSide,
                })}
              >{`${playSide}P`}</a>
            </List.Item>
          ))
        ) : (
          <List.Item>
            <a
              href={generateTextageURL(music, map, {
                playStyle: PlayStyle.Dp,
              })}
            >
              DP
            </a>
          </List.Item>
        )}
      </List>
    </div>
  </>
)

export default MapTitleHeader
