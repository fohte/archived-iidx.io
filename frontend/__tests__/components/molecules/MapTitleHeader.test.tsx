import * as _ from 'lodash'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import MapTitleHeader, { Props } from '@app/components/molecules/MapTitleHeader'
import { Difficulty, PlayStyle } from '@app/queries'

describe('MapTitleHeader', () => {
  const makeMusic = (music?: Partial<Props['music']>): Props['music'] => ({
    title: 'title',
    subTitle: 'subTitle',
    genre: 'genre',
    artist: 'artist',
    series: 1,
    textageUid: 'textageUid',
    ...music,
  })

  const makeMap = (map?: Partial<Props['map']>): Props['map'] => ({
    level: 12,
    playStyle: PlayStyle.Sp,
    difficulty: Difficulty.Another,
    minBpm: 100,
    maxBpm: 100,
    numNotes: 1000,
    ...map,
  })

  const cases = [
    { playStyle: PlayStyle.Sp, minBpm: 100, maxBpm: 100 },
    { playStyle: PlayStyle.Sp, minBpm: 100, maxBpm: 400 },
    { playStyle: PlayStyle.Dp, minBpm: 100, maxBpm: 100 },
    { playStyle: PlayStyle.Dp, minBpm: 100, maxBpm: 400 },
  ]

  cases.forEach(({ playStyle, minBpm, maxBpm }) => {
    it(`renders correctly when playStyle is ${playStyle} and BPM is ${minBpm}-${maxBpm}`, () => {
      const props: Props = {
        music: makeMusic(),
        map: makeMap({ playStyle, minBpm, maxBpm }),
      }

      const tree = renderer.create(<MapTitleHeader {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
