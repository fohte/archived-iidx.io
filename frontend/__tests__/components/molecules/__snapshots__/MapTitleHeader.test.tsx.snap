// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapTitleHeader renders correctly when playStyle is DP and BPM is 100-100 1`] = `
Array [
  <div>
    genre
    <h2
      className="ui header"
    >
      title
       
      subTitle
    </h2>
    artist
  </div>,
  <div>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        DP
         
        ANOTHER
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        ☆
        12
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        BPM 
        100
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        1000
         notes
      </div>
    </div>
  </div>,
  <div>
    <span
      style={
        Object {
          "marginRight": "0.5em",
        }
      }
    >
      TexTage:
    </span>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <a
          href="//textage.cc/score/1/textageUid.html?DAC00"
        >
          DP
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`MapTitleHeader renders correctly when playStyle is DP and BPM is 100-400 1`] = `
Array [
  <div>
    genre
    <h2
      className="ui header"
    >
      title
       
      subTitle
    </h2>
    artist
  </div>,
  <div>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        DP
         
        ANOTHER
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        ☆
        12
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        BPM 
        100-400
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        1000
         notes
      </div>
    </div>
  </div>,
  <div>
    <span
      style={
        Object {
          "marginRight": "0.5em",
        }
      }
    >
      TexTage:
    </span>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <a
          href="//textage.cc/score/1/textageUid.html?DAC00"
        >
          DP
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`MapTitleHeader renders correctly when playStyle is SP and BPM is 100-100 1`] = `
Array [
  <div>
    genre
    <h2
      className="ui header"
    >
      title
       
      subTitle
    </h2>
    artist
  </div>,
  <div>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        SP
         
        ANOTHER
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        ☆
        12
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        BPM 
        100
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        1000
         notes
      </div>
    </div>
  </div>,
  <div>
    <span
      style={
        Object {
          "marginRight": "0.5em",
        }
      }
    >
      TexTage:
    </span>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <a
          href="//textage.cc/score/1/textageUid.html?1AC00"
        >
          1P
        </a>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <a
          href="//textage.cc/score/1/textageUid.html?2AC00"
        >
          2P
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`MapTitleHeader renders correctly when playStyle is SP and BPM is 100-400 1`] = `
Array [
  <div>
    genre
    <h2
      className="ui header"
    >
      title
       
      subTitle
    </h2>
    artist
  </div>,
  <div>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        SP
         
        ANOTHER
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        ☆
        12
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        BPM 
        100-400
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        1000
         notes
      </div>
    </div>
  </div>,
  <div>
    <span
      style={
        Object {
          "marginRight": "0.5em",
        }
      }
    >
      TexTage:
    </span>
    <div
      className="ui divided horizontal list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <a
          href="//textage.cc/score/1/textageUid.html?1AC00"
        >
          1P
        </a>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <a
          href="//textage.cc/score/1/textageUid.html?2AC00"
        >
          2P
        </a>
      </div>
    </div>
  </div>,
]
`;
